gems/sorbet-runtime/lib/types/props/serializable.rb

```diff
@@ -310,9 +159,45 @@ def add_prop_definition(prop, rules)
     rules[:serialized_form] = rules.fetch(:name, prop.to_s)
     res = super
     prop_by_serialized_forms[rules[:serialized_form]] = prop
+    enqueue_lazy_method_definition!(:__t_props_generated_serialize) {generate_serialize_source}
+    enqueue_lazy_method_definition!(:__t_props_generated_deserialize) {generate_deserialize_source}
     res
   end
 
+  private def generate_serialize_source
+    T::Props::Private::SerializerGenerator.generate(props)
+  end
+
+  private def generate_deserialize_source
+    T::Props::Private::DeserializerGenerator.generate(
+      props,
+      props_with_defaults || {},
+    )
+  end
+
+  def raise_nil_deserialize_error(hkey)
```

Comment:
This is pulled out from `T::Props::Serializable#deserialize` so that we don't have to codegen it